### PR TITLE
WIP: SidebarPage Content element

### PR DESCRIPTION
**Pull request by Sarabadu**
## Hey, I just made a Pull Request!

Creates a Sidebar content Element and stores a ref to that element in a context and creates a "skip to content" hidden button to follow this [accessibility recommendation](https://www.w3.org/TR/WCAG20-TECHS/G1.html) 

#### Accessibility Demo
https://user-images.githubusercontent.com/1770453/135339890-08b318c1-d06b-48ab-be3a-691f9875f86c.mp4

#### Focus content
This also enables the possibility to set focus directly on the page content element using the `useContentRef` hook and  `contentRef.current.focus()`

### TODO
- [x] Test if there are no breaking changes
- [x] Update the example app
- [x] Update docs (storybook)

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/core-components/api-report.md [hunk 1]
```diff
@@ -928,6 +928,23 @@ export const SidebarItem: React_2.ForwardRefExoticComponent<
...
 
+// Warning: (ae-missing-release-tag) "SideBarPageContent" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
+//
+// @public (undocumented)
+export function SideBarPageContent(props: PropsWithChildren<{}>): JSX.Element;
+
+// Warning: (ae-missing-release-tag) "SidebarPageContext" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
+//
+// @public (undocumented)
+export const SidebarPageContext: React_2.Context<SidebarPageContextType>;
```
Comment (Rugvip) [inline packages/core-components/api-report.md:939]: Does this need to be exported?

**Review by Rugvip — COMMENTED** [review]
Thanks!

The `SideBarPageContent` is new public API right? How is intended to be used? Best add it to the example app, docs, or tests so that we can see the intended usage in the PR

File: packages/core-components/api-report.md (hunk 1 above)
Comment (Sarabadu) [reply packages/core-components/api-report.md:939]: That export is intentional, to have a practical was to focus on the mail content from any other component.

**Comment by Sarabadu** [comment]
> Thanks!
> 
> The `SideBarPageContent` is new public API right? How is intended to be used? Best add it to the example app, docs, or tests so that we can see the intended usage in the PR

Thanks, updated the TODO on the description

**Comment by Rugvip** [comment]
I'm still not 100% sure what the purpose is tbh, but is this something that could be baked into `SidebarPage` and perhaps the `Sidebar` component?

**Comment by Sarabadu** [comment]
I've updated the Storybook and the example app to show the intended usage

**Pull request closed without merging**